### PR TITLE
Fixing "null" entry in test.xml with specs2

### LINKED ISSUE #231: null entry in testlog xml
May cause problems with parsers.

E.g., the first entry here:
```
<?xml version='1.0' encoding='UTF-8'?>
<testsuites>
  <testsuite name='null' timestamp='1970-04-19T02:06:01.847Z' hostname='localhost' tests='3' failures='0' errors='0' time='0.284' package='' id='0'>
    <properties />
    <system-out />
    <system-err />
  </testsuite>
  <testsuite name='com.example.SomeJunit2Test' timestamp='1970-04-19T02:06:01.847Z' hostname='localhost' tests='1' failures='0' errors='0' time='0.004' package='' id='1'>
    <properties />
    <testcase name='passingTest2' classname='com.example.SomeJunit2Test' time='0.004' />
    <system-out />
    <system-err />
  </testsuite>
  <testsuite name='com.example.SomeJunitTest' timestamp='1970-04-19T02:06:01.851Z' hostname='localhost' tests='1' failures='0' errors='0' time='0.0' package='' id='2'>
    <properties />
    <testcase name='passingTest' classname='com.example.SomeJunitTest' time='0.0' />
    <system-out />
    <system-err />
  </testsuite>
  <testsuite name='SomeSpecs2Test' timestamp='1970-04-19T02:06:02.131Z' hostname='localhost' tests='1' failures='0' errors='0' time='0.0' package='' id='3'>
    <properties />
    <system-out />
    <system-err />
  </testsuite>
  <testsuite name='example test should' timestamp='1970-04-19T02:06:02.131Z' hostname='localhost' tests='1' failures='0' errors='0' time='0.0' package='' id='4'>
    <properties />
    <testcase name='example test should::allow to pick only me' classname='com.example.SomeSpecs2Test' time='0.0' />
    <system-out />
    <system-err />
  </testsuite>
</testsuites>
```

### PR DESCRIPTION
**Pull request by hmemcpy**
Additionally, filtering out runners that are not actually run, to reduce the amount of empty entries.
This fixes #231

### CODE REVIEW

File: src/java/io/bazel/rulesscala/test_discovery/FilteredRunnerBuilder.scala [hunk 1]
```diff
@@ -11,17 +11,14 @@ import scala.collection.JavaConverters._
 
 class FilteredRunnerBuilder(builder: RunnerBuilder, filteringRunnerBuilder: FilteringRunnerBuilder) extends RunnerBuilder {
   // Defined by --test_filter bazel flag.
-  private val maybePattern = sys.env.get("TESTBRIDGE_TEST_ONLY").map(Pattern.compile)
-
-  def runnerFilter(runner: Runner, testClass: Class[_], pattern: Pattern): Runner = runner
```
Comment (ittaiz) [inline src/java/io/bazel/rulesscala/test_discovery/FilteredRunnerBuilder.scala]: I think I’m missing something.
Why was this removed?
Also, why did you add type annotation to the below line?

Comment (hmemcpy) [reply src/java/io/bazel/rulesscala/test_discovery/FilteredRunnerBuilder.scala]: This function wasn't used anywhere, I don't even know why it was added in the first place.
And the type annotation is a leftover, I can remove it.

**Comment by hmemcpy** [comment]
An example of this fix. Suppose we're running with `--test_filter=com.example.SomeSpecs2Test#`

Before:
```
<?xml version='1.0' encoding='UTF-8'?>
<testsuites>
  <testsuite name='null' timestamp='1970-01-01T07:18:56.036Z' hostname='localhost' tests='14' failures='0' errors='0' time='0.056' package='' id='0'>
    <properties />
    <system-out />
    <system-err />
  </testsuite>
  <testsuite name='SomeSpecs2Test' timestamp='1970-01-01T07:18:56.036Z' hostname='localhost' tests='7' failures='0' errors='0' time='0.056' package='' id='1'>
    <properties />
    <system-out />
    <system-err />
  </testsuite>
  <testsuite name='SomeSpecs2Test should' timestamp='1970-01-01T07:18:56.036Z' hostname='localhost' tests='7' failures='0' errors='0' time='0.056' package='' id='2'>
    <properties />
    <testcase name='SomeSpecs2Test should::do something' classname='com.example.SomeSpecs2Test' time='0.003' />
    ...
    <system-out />
    <system-err />
  </testsuite>
  <testsuite name='UnrelatedTest' timestamp='' hostname='localhost' tests='6' failures='0' errors='0' time='0.0' package='' id='3'>
    <properties />
    <system-out />
    <system-err />
  </testsuite>
  <testsuite name='Unrelated test should' timestamp='' hostname='localhost' tests='6' failures='0' errors='0' time='0.0' package='' id='4'>
    <properties />
    <system-out />
    <system-err />
  </testsuite>
  <testsuite name='AnotherUnrelatedTest' timestamp='' hostname='localhost' tests='1' failures='0' errors='0' time='0.0' package='' id='5'>
    <properties />
    <system-out />
    <system-err />
  </testsuite>
  <testsuite name='AnotherUnrelated test should' timestamp='' hostname='localhost' tests='1' failures='0' errors='0' time='0.0' package='' id='6'>
    <properties />
    <system-out />
    <system-err />
  </testsuite>
</testsuites>
```
After:
```
<?xml version='1.0' encoding='UTF-8'?>
<testsuites>
  <testsuite name='Aggregate Specs2 Test Suite' timestamp='1970-01-01T07:16:30.809Z' hostname='localhost' tests='7' failures='0' errors='0' time='0.062' package='' id='0'>
    <properties />
    <system-out />
    <system-err />
  </testsuite>
  <testsuite name='SomeSpecs2Test' timestamp='1970-01-01T07:16:30.809Z' hostname='localhost' tests='7' failures='0' errors='0' time='0.062' package='' id='1'>
    <properties />
    <system-out />
    <system-err />
  </testsuite>
  <testsuite name='SomeSpecs2Test should' timestamp='1970-01-01T07:16:30.809Z' hostname='localhost' tests='7' failures='0' errors='0' time='0.062' package='' id='2'>
    <properties />
    <testcase name='SomeSpecs2Test should::do something' classname='com.example.SomeSpecs2Test' time='0.002' />
    ...
    <system-out />
    <system-err />
  </testsuite>
</testsuites>
```

Notice that the `null` and the irrelevant tests no longer show.

**Comment by ittaiz** [comment]
Can you read a bit about this method? I think it comes from (and maybe also
used by) Junit

**Comment by ittaiz** [comment]
you have failing tests

**Pull request merged**